client/lib/domains/resolve-domain-status.tsx

```diff
@@ -17,18 +18,47 @@ import {
 	domainMappingSetup,
 } from 'calypso/my-sites/domains/paths';
 import { transferStatus, type as domainTypes, gdprConsentStatus } from './constants';
+import type { ResponseDomain } from './types';
+import type { ReactChild } from 'react';
+
+export type ResolveDomainStatusReturn =
+	| {
+			statusText: ReactChild | Array< ReactChild >;
+			statusClass:
+				| 'status-error'
+				| 'status-warning'
+				| 'status-alert'
+				| 'status-success'
+				| 'status-neutral'
+				| 'status-premium';
+			status: ReactChild;
+			icon: 'info' | 'verifying' | 'check_circle' | 'cached' | 'cloud_upload' | 'download_done';
+			listStatusText?: ReactChild | Array< ReactChild >;
+			listStatusClass?: 'alert' | 'warning' | 'verifying' | 'info' | 'premium' | 'transfer-warning';
+			listStatusWeight?: number;
+			noticeText?: ReactChild | Array< ReactChild > | null;
+	  }
+	| Record< string, never >;
```

Comment:
I had to research this `Record< string, never >` for a bit 😄 does this mean that the only object that will be returned is the one defined there? And all other returned objects that do not conform with the type definition will throw an exception? In that case, wouldn't just defining the return type be enough?

Reply:
ah - `Record<string, never>` is used for when you want to return an empty object. if none of the rules matches the function would return an empty object - I'm not sure this ever happens but since that's how the code works I preferred to keep it as is

Reply:
Oh I see, that makes sense! An empty object is the only one that conforms to that type.